src/compile.c

```diff
@@ -997,16 +998,52 @@ static int count_cfunctions(block b) {
   return n;
 }
 
+#ifdef WIN32
+extern const char **environ;
+#endif
+
+static jv env = {JV_KIND_INVALID, 0, 0, 0, {0}};
+
+static void
+free_env(void)
+{
+  jv_free(env);
+}
+
+static jv
+make_env(void)
+{
+  if (jv_is_valid(env))
+    return jv_copy(env);
+  jv r = jv_object();
+  if (environ == NULL)
+    return r;
+  for (size_t i = 0; environ[i] != NULL; i++) {
+    const char *eq;
+
+    if ((eq = strchr(environ[i], '=')) == NULL)
+      r = jv_object_delete(r, jv_string(environ[i]));
+    else
+      r = jv_object_set(r, jv_string_sized(environ[i], eq - environ[i]), jv_string(eq + 1));
+  }
+  atexit(free_env);
```

Comment:
Actually, I'm unhappy with the use of atexit() for this.  I think I'd rather pass `$ARGS` all the way down here and output `$ENV` so it can be freed after compilation.